### ISSUE TITLE
Open Closed Principle

### ISSUE THREAD

**Issue by s72785** [SemAhto/WordCloudCalculatorModule#2]
@vardoo Wie setzen wir das um?
Wird dann die Klasse `Program` einfach für abstrakt erklärt und mehrere Implementationen stehen neben einander…

**Comment by kayak194** [comment]
Nein, die Klasse Program in der ConsoleApplication ist nur zum Testen und freuen dass es funktioniert xD. Euer Ansatzpunkt wäre das Interface `WordCloudCalculator.ExtractingWordCloudCalculator.IWordAppearenceCalculationMethod` neu zu implementieren.

**Comment by s72785** [comment]
Das heißt jetzt konkret "überschreiben" (wie in Java mit `@override`)?

**Comment by kayak194** [comment]
Ein konkretes Beispiel ist `WordCloudCalculator.ExtractingWordCloudCalculator.SimpleAppearenceCalculationMethod`. Deins legst du daneben. In der Klasse `Program` ist  `SimpleAppearenceCalculationMethod` als Typ Parameter an `ExtractingWorfCloudCalculator` übergeben worden. Das mit euer neuen Implementierung ersetzen und freuen xD.

**Comment by s72785** [comment]
Ok, danke erstmal.
